src/dev_routing.js

```diff
@@ -0,0 +1,65 @@
+import {Router, Route, IndexRedirect, Link, browserHistory} from 'react-router';
+
+import React from 'react';
+import App from './App';
+import Login from './Login';
+import Register from './Register';
+
+import query from './util/query';
+
+function LoginPage () {
+    return (
+        <div>
+            <Link to="/register">Register</Link>
+            <Login />
+        </div>
+    );
+}
+
+function RegisterPage () {
+    return (
+        <div>
+            <Link to="/login">Login</Link>
+            <Register />
+        </div>
+    );
+}
+
+const logout = async () => {
+    await fetch('/logout');
+    window.location = '/login';
+};
+
+function MainPage () {
+    return (
+        <div>
+            <div>Logged in!</div>
+            <button onClick={logout}>Logout</button>
+        </div>
+    );
+}
+
+async function requireAuth (nextState, replace, next) {
+    const data = await query({query: '{ auth }'});
```

Comment:
In a single page app, we shouldn't need to check with the server before transition. This is fundamentally broken.